test/questions_test.py

```diff
@@ -19,31 +19,36 @@ def test_no_facets_then_return_no_questions(self):
 
     def test_one_discriminating_facet_then_return_one_element_question(self):
         question_generator = QuestionGenerator()
-        facet_a = Facet('NameA', 'ValueA')
+        facet_a1 = Facet('NameA', 'ValueA1')
+        facet_a2 = Facet('NameA', 'ValueA2')
         facet_b = Facet('NameB', 'ValueB')
+        facets_by_document = {'1': [facet_a1], '2': [facet_a2], '3': [facet_a2], '4': [facet_b]}
```

Comment:
Here the goal of the test is to return a one element question, So a better option would be:
`facet_a = Facet('NameA', 'ValueA')`
`facet_b = Facet('NameB', 'ValueB')`
`facets_by_document = {'1': [facet_a, facet_b], '2': [facet_a], '3': [facet_a]}`
Would probably be simpler to read and understand that way

Reply:
the facet NameA must have 2 values to be considered as discriminating. We absolutely need ValueA1 and ValueA2 . But facet_b is optional.